### ISSUE TITLE
[Bug] false error on magnet links

### ISSUE THREAD

**Issue by milahu** [theoludwig/markdownlint-rule-relative-links#1]
bug: magnet links are treated as relative links

## Steps To Reproduce

```md
[some magnet link](magnet:?xt=urn:btih:asdf)
```

## The current behavior

```
relative-links Relative links should be valid [Link "magnet:?xt=urn:btih:asdf" is dead]
```

currently, some external protcols are hard-coded

```js
const EXTERNAL_PROTOCOLS = new Set([
  'http:',
  'https:',
  'mailto:',
  'tel:',
  'ftp:'
])
```

```js
        if (hrefSrc != null) {
          const url = new URL(hrefSrc, pathToFileURL(params.name))
          url.hash = ''
          const isRelative =
            hrefSrc.startsWith('./') ||
            hrefSrc.startsWith('../') ||
            !EXTERNAL_PROTOCOLS.has(url.protocol)
          if (isRelative && !fs.existsSync(url.pathname)) {
            const detail = `Link "${hrefSrc}" is dead`
            addError(onError, lineNumber, detail)
          }
        }
```

## The expected behavior

it should ignore magnet links

more generally, it should ignore all links with protocol except the `file:` protocol

```console
$ node
> new URL("magnet:?xt=urn:btih:asdf").protocol
'magnet:'
> new URL("file:asdf").protocol
'file:'
> new URL("file:///asdf").protocol
'file:'
> new URL("asdf").protocol
Uncaught TypeError [ERR_INVALID_URL]: Invalid URL
> new URL("./asdf").protocol
Uncaught TypeError [ERR_INVALID_URL]: Invalid URL
```

**Comment by theoludwig** [comment]
Hey! @milahu 
Thanks for your report, this is indeed a bug!

Yes I think, it can be a better solution to check the `file:` protocol instead to rely on a `Set` hard-coded in the code. :+1:
Are you willing to make a PR to fix this? :smile:

**Comment by milahu** [comment]
nope, sorry

**Comment by theoludwig** [comment]
Alright, no problem.
I fixed it in https://github.com/Divlo/markdownlint-rule-relative-links/commit/b6b092dc1fc416d0abb0060636e2d56a6a60b2b2.
